src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -2979,85 +2974,73 @@ class CgroupUtils:
         for attempt in range(2):
             avail_resc = self.available_node_resources(node)
             assigned = self.assign_job(hostresc, avail_resc, node)
-            if not assigned and (attempt < 3):
+            if not assigned:
                 # No resources were assigned to the job.
                 # Most likely cause was that a cgroup has
                 # not been cleaned up yet
-                pbs.logmsg(pbs.EVENT_DEBUG2, "Failed to assign resources. " +
-                           "Checking the cgroups on the node " +
-                           "with the server")
-                # Collect the jobs on the node (try reading mom_priv/jobs,
-                # if not successful ask server)
+                pbs.logmsg(pbs.EVENT_DEBUG2, "Failed to assign job resources")
+                pbs.logmsg(pbs.EVENT_DEBUG2, "Resyncing local job data")
+                # Collect the jobs on the node (try reading mom_priv/jobs)
                 jobs_list = []
-                if attempt < 2:
-                    jobs_list = node.gather_jobs_on_node_local()
-                else:
-                    # If after the first attempt we are still having issues
-                    # look at the server jobs list instead of the moms
+                try:
                     jobs_list = node.gather_jobs_on_node()
-                if jobs_list is not None:
-                    # Don't clean up the cgroup we just made for the
-                    # job just yet
-                    if jobid not in jobs_list:
-                        jobs_list.append(jobid)
+                except:
+                    pbs.logmsg(pbs.EVENT_DEBUG2,
+                               "Failed to resyncing local job data")
+                # Job list should contain the new jobid. Do not attempt to
+                # cleanup orphaned cgroups with an incomplete job list.
+                # There could be other active jobs missing from the list.
+                if jobs_list and jobid not in jobs_list:
+                    pbs.logmsg(pbs.EVENT_DEBUG2, "Job not found: %s" % jobid)
+                else:
                     self.cleanup_orphans(jobs_list)
+                # Pause after the first attempt
+                if attempt < 1:
+                    sleep(0.5)
         if not assigned:
-            # Cleanup cgroups for jobs not present on this node
+            # Log a message and rerun the job
             pbs.logmsg(pbs.EVENT_DEBUG2,
-                       "Assign resources failed. Attempting to cleanup all " +
-                       "leftover cgroups (including event job %s)" % jobid)
-            if not jobs_list:
-                pbs.logmsg(pbs.EVENT_DEBUG2, "Local job list is empty!")
-            else:
-                # Remove the current job from the jobs list so it will be
-                # cleaned up as well.
-                try:
-                    jobs_list.remove(jobid)
-                    self.cleanup_orphans(jobs_list)
```

Comment:
That's part of the periodic handler now. Code was also added to remove a cgroup if it is already present when the execjob_begin handler is called.